### ISSUE TITLE
torch.nn.ReplicationPad{1|2}d supports more input dimension than are written on documentation

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#83234]
### 📚 The doc issue

torch.nn.ReplicationPad1d supports (C,W) or (N, C, W) input, and needs a int or 2-tuple padding on the documentation. That is to say, torch.nn.ReplicationPad1d only supports 2D or 3D tensor.

torch.nn.ReplicationPad2d supports (C,H,W) or (N, C, H,W) input, and needs a int or 4-tuple padding on the documentation. That is to say, torch.nn.ReplicationPad2d only supports 3D or 4D tensor.

After running code, I find that torch.nn.ReplicationPad1d also supports 4D tensor with 4-tuple padding, and torch.nn.ReplicationPad2d also supports 5D tensor with 6-tuple padding.

```
import torch
results={}
arg_1 = [3,0,2,1]
arg_class = torch.nn.ReplicationPad1d(arg_1,)
arg_2 = torch.rand([25, 2, 46, 1], dtype=torch.float32)
results['res'] = arg_class(arg_2)
```

```
import torch
results={}
arg_1 = [2,2,2,2,2,2]
arg_class = torch.nn.ReplicationPad2d(arg_1,)
arg_2 = torch.rand([4, 1, 1, 3, 3], dtype=torch.float32)
results['res'] = arg_class(arg_2)
```
Above code runs well.

**Comment by cheyennee** [comment]
I found the same problem in torch.nn.ZeroPad2d. torch.nn.ZeroPad2d supports (C,H,W) or (N, C, H,W) input, and needs a int or 4-tuple padding on the documentation. That is to say, torch.nn.ReplicationPad2d only supports 3D or 4D tensor.

After running code, I find that torch.nn.ZeroPad2d also supports 5D tensor with 4-tuple padding.

```
import torch
results={}
arg_1 = [1, 1, 1, 1]
arg_class = torch.nn.ZeroPad2d(arg_1,)
arg_2 = torch.rand([1, 1, 1, 3, 3], dtype=torch.float32)
results['res'] = arg_class(arg_2)
```
